### PR TITLE
Add go list check in CI

### PR DESCRIPTION
**Pull request by mlguerrero12**

### CODE REVIEW

File: tasks.py [hunk 1]
```diff
@@ -889,6 +889,13 @@ def gomodtidy(ctx):
     if not res.ok:
         raise Exit(message="go mod tidy failed")
 
+@task
+def golist(ctx):
+    """Runs go list"""
+    res = run("go list -modfile=go.mod -m -json -mod=mod all", hide="out")
```
Comment (fedepaol) [inline tasks.py:895]: I'd remove the -json flag here

Comment (mlguerrero12) [reply tasks.py:895]: done

**Comment by mlguerrero12** [comment]
@fedepaol, any idea why this is failing? This doesn't seem to be correct.

#### Uncommitted files found, you may need to run inv generatemanifests ####

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -78,6 +78,7 @@ jobs:
           inv verifylicense
           inv gomodtidy
           inv checkchanges --action="run inv gomodtidy"
+          inv golist
```
Comment (fedepaol) [inline .github/workflows/ci.yaml:81]: you must follow this with an `inv checkchanges --action="run inv golist"`

**Comment by fedepaol** [comment]
> @fedepaol, any idea why this is failing? This doesn't seem to be correct.
> 
> #### Uncommitted files found, you may need to run inv generatemanifests

The comment is wrong because we missed a check (left a comment). The output shows that go.sum in not committed (and if I try to run inv golist locally it shows a changed go.sum)

**Comment by mlguerrero12** [comment]
> > @fedepaol, any idea why this is failing? This doesn't seem to be correct.
> > #### Uncommitted files found, you may need to run inv generatemanifests
> 
> The comment is wrong because we missed a check (left a comment). The output shows that go.sum in not committed (and if I try to run inv golist locally it shows a changed go.sum)

Go.sum didn't change for me. Will check again. Thanks.

**Comment by mlguerrero12** [comment]
Still the same. However, when I deleted the content of the go.sum and ran again inv golist, there were some changes. Let's see.

**Pull request closed without merging**